src/tree/param.h

```diff
@@ -99,26 +90,6 @@ struct TrainParam : public dmlc::Parameter<TrainParam> {
         .describe(
             "Maximum depth of the tree; 0 indicates no limit; a limit is required "
             "for depthwise policy");
-    DMLC_DECLARE_FIELD(max_leaves).set_lower_bound(0).set_default(0).describe(
-        "Maximum number of leaves; 0 indicates no limit.");
-    DMLC_DECLARE_FIELD(max_bin).set_lower_bound(2).set_default(256).describe(
```

Comment:
Oops, my bad. I had erroneously assumed that the parameter was only used by the fast histogram updater. I've put back `max_bins` and other parameters back to `TrainParam`.